x-pack/plugin/esql/compute/test/src/main/java/org/elasticsearch/compute/test/AnyOperatorTestCase.java

```diff
@@ -32,11 +32,27 @@
  * Superclass for testing any {@link Operator}, including {@link SourceOperator}s.
  */
 public abstract class AnyOperatorTestCase extends ComputeTestCase {
+    /**
+     * @param requiresDeterministicFactory
+     *          True if the returned {@link Operator.OperatorFactory} should always generate an identical deterministic operator.
+     *          That is, for two different calls, both operators should do "exactly" the same.
+     */
+    protected record SimpleOptions(boolean requiresDeterministicFactory) {
+        public static final SimpleOptions DEFAULT = new SimpleOptions(false);
+    }
```

Comment:
Created this structure instead of just a boolean as it's:
- Easier to understand from callers
- Extendable (Not a big concern now, but the first point makes more sense to me)

Reply:
:+1: